nullaway/src/main/java/com/uber/nullaway/generics/GenericsChecks.java

```diff
@@ -46,8 +46,7 @@
 /** Methods for performing checks related to generic types and nullability. */
 public final class GenericsChecks {
 
-  /** Do not instantiate; all methods should be static */
-  private GenericsChecks() {}
+  private final Map<Tree, Map<Type, Type>> inferredTypes = new HashMap<>();
```

Comment:
Add Javadoc on what this is.  For now, can the key type for the outer map be `MethodInvocationTree`?  And are the keys of the nested maps always `TypeVariable`s?